### PR TITLE
feat: include all relevant CF-Groups (and marked CFs) for TRaSH-Guide…

### LINKED ISSUE #302: refactor: adjust required loading of cf groups with TRaSH-Guide changes
- changes to the HDR formats will require to load CF groups automatically without user defining them to keep the old behaviour intact.
- For the future the CF-groups will probably be mandatory to be read for TRaSH-Guide profiles

### PR DESCRIPTION
**Pull request by BlackDark**
… profiles automatically

* This will required and defaults CFs defined in the groups always for the TRaSH-Guide profiles
* now all cf-groups will be loaded for TRaSH-Guide templates/profiles
* the CFs in the profiles will be mostly moved from the profiles into groupes where the profiles are referenced
* see https://github.com/TRaSH-Guides/Guides/pull/2455#discussion_r2297832409
* see https://discord.com/channels/492590071455940612/1409911784386596894
* Closes #302 

## Summary by Sourcery

Support automatic inclusion of relevant CF-Groups in TRaSH-Guide profiles by filtering on default/required flags, respecting profile exclusions, integrating the new logic into the template importer, and updating types, tests, and documentation accordingly

New Features:
- Automatically include custom formats from CF-Groups for TRaSH-Guide profiles via a new transformTrashQPCFGroups function
- Integrate CF-Group inclusion into the template import process to load default CF-Groups and assign scores to profiles
- Extend configuration types to capture default flags and optional descriptions for CF-Groups

Documentation:
- Update configuration documentation to explain automatic CF-Group inclusion and link relevant TRaSH-Guides resources

Tests:
- Add unit tests for transformTrashQPCFGroups covering default flags, required flags, profile exclusions, multiple groups, and edge cases

### CODE REVIEW

File: src/trash-guide.ts [hunk 1]
```diff
@@ -319,6 +319,40 @@ export const transformTrashQPCFs = (data: TrashQP): ConfigCustomFormat => {
...
+  // Traverse each CF group and check for default=true
+  for (const [_, cfGroup] of trashCFGroupMapping) {
+    // Check if the default prop is truthy (string "true")
+    if (cfGroup.default === "true") {
+      // Check if template is excluded via exclude field
+      const isExcluded = cfGroup.quality_profiles?.exclude?.[profileName] != null;
+
+      if (!isExcluded) {
+        // Include all required and default CFs from this group
+        const cfsToInclude = cfGroup.custom_formats.filter((cf) => cf.required || cf.default === true);
```
[the remark this answers is not in the document]
Comment (BlackDark) [reply src/trash-guide.ts:335]: The "default" prop in custom_formats should only be booleans. only on the group level the "default" prop is a string type
